src/posix/platform/trel_udp6.cpp

```diff
@@ -141,60 +137,20 @@ static const char *BufferToString(const uint8_t *aBuffer, uint16_t aLength)
 
 #if OPENTHREAD_CONFIG_POSIX_TREL_USE_NETLINK_SOCKET
 
-static void UpdateUnicastAddress(const otIp6Address *aUnicastAddress, bool aToAdd)
-{
-    int            netlinkSocket;
-    int            ret;
-    struct rtattr *rta;
-
-    struct
-    {
-        struct nlmsghdr  nh;
-        struct ifaddrmsg ifa;
-        char             buf[64];
-    } request;
-
-    netlinkSocket = SocketWithCloseExec(AF_NETLINK, SOCK_DGRAM, NETLINK_ROUTE, kSocketNonBlock);
-    VerifyOrDie(netlinkSocket >= 0, OT_EXIT_ERROR_ERRNO);
-
-    memset(&request, 0, sizeof(request));
-
-    request.nh.nlmsg_len   = NLMSG_LENGTH(sizeof(struct ifaddrmsg));
-    request.nh.nlmsg_flags = NLM_F_REQUEST | NLM_F_ACK | NLM_F_CREATE | NLM_F_EXCL;
-    request.nh.nlmsg_type  = aToAdd ? RTM_NEWADDR : RTM_DELADDR;
-    request.nh.nlmsg_pid   = 0;
-    request.nh.nlmsg_seq   = 0;
-
-    request.ifa.ifa_family    = AF_INET6;
-    request.ifa.ifa_prefixlen = TREL_UNICAST_ADDRESS_PREFIX_LEN;
-    request.ifa.ifa_flags     = IFA_F_NODAD;
-    request.ifa.ifa_scope     = TREL_UNICAST_ADDRESS_SCOPE;
-    request.ifa.ifa_index     = (unsigned int)(sInterfaceIndex);
-
-    rta = reinterpret_cast<struct rtattr *>((reinterpret_cast<char *>(&request)) + NLMSG_ALIGN(request.nh.nlmsg_len));
-    rta->rta_type = IFA_LOCAL;
-    rta->rta_len  = RTA_LENGTH(sizeof(otIp6Address));
-
-    memcpy(RTA_DATA(rta), aUnicastAddress, sizeof(otIp6Address));
-
-    request.nh.nlmsg_len = NLMSG_ALIGN(request.nh.nlmsg_len) + rta->rta_len;
-
-    ret = send(netlinkSocket, &request, request.nh.nlmsg_len, 0);
-    VerifyOrDie(ret != -1, OT_EXIT_ERROR_ERRNO);
-
-    close(netlinkSocket);
-}
-
 static void AddUnicastAddress(const otIp6Address *aUnicastAddress)
 {
     otLogDebgPlat("[trel] AddUnicastAddress(%s)", Ip6AddrToString(aUnicastAddress));
-    UpdateUnicastAddress(aUnicastAddress, /* aToAdd */ true);
+    ot::Posix::NetifManager::Get().AddUnicast(
+        sInterfaceIndex,
+        otIp6AddressInfo{aUnicastAddress, TREL_UNICAST_ADDRESS_PREFIX_LEN, TREL_UNICAST_ADDRESS_SCOPE, true});
 }
 
 static void RemoveUnicastAddress(const otIp6Address *aUnicastAddress)
 {
     otLogDebgPlat("[trel] RemoveUnicastAddress(%s)", Ip6AddrToString(aUnicastAddress));
-    UpdateUnicastAddress(aUnicastAddress, /* aToAdd */ false);
+    ot::Posix::NetifManager::Get().RemoveUnicast(
+        sInterfaceIndex,
+        otIp6AddressInfo{aUnicastAddress, TREL_UNICAST_ADDRESS_PREFIX_LEN, TREL_UNICAST_ADDRESS_SCOPE, true});
```

Comment:
Thanks @bukepo  for this PR. I think refactoring the netlink and netif manager into their own module that is then used by different platform modules is indeed very good idea.

But there is one thing that makes be feel a bit uneasy about  change to the code in the trel posix platform code. Since https://github.com/openthread/openthread/issues/6472 we are no longer covering the trel posix behavior in any CI test and if anything gets broken we won't be detecting it.  I feel much better if we leave the `trel_udp6.cpp` as is for now, and change the code here later after some test coverage is added back. Thoughts?